File: bench/bench_array.py
import time
N = 1_000_000
t0 = time.process_time()
arr = [0] * (N+1)
for i in range(1, N+1): arr[i] = i
s = 0
for i in range(1, N+1): s += arr[i]
t1 = time.process_time()
print(f"python array 1M = {s}  time: {t1-t0:.4f}s")

File: bench/bench_array.java
class bench_array {
    static long run(int n) {
        long[] arr = new long[n + 1];
        for (int i = 1; i <= n; i++) arr[i] = i;
        long s = 0;
        for (int i = 1; i <= n; i++) s += arr[i];
        return s;
    }

    public static void main(String[] args) {
        for (int w = 0; w < 3; w++) run(100_000); // warm-up, discarded
        long t0 = cpu();
        long s = run(1_000_000);
        long t1 = cpu();
        System.out.printf("java   array 1M = %d  time: %.4fs%n", s, (t1 - t0) / 1e9);
    }

    static long cpu() {
        return java.lang.management.ManagementFactory.getThreadMXBean().getCurrentThreadCpuTime();
    }
}
